### PR TITLE
core: update golang version to 1.20

### PR DESCRIPTION
**Pull request by subhamkrai**
core: update golang version to 1.20

Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

**Comment by subhamkrai** [comment]
> Looks like there is an issue with the CI. Maybe a timeout is needed since they have been running for four hours.

yeah, we can add the meantime #142 will fix this CI issue once that is merged it will be green again and we can add timeout in follow-up PR.

File: .github/workflows/go-test.yaml [hunk 1]
```diff
@@ -131,6 +131,10 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
+      - name: setup golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: "1.20"
```
Comment (BlaineEXE) [inline .github/workflows/go-test.yaml:137]: Did you want to keep this step with the one above on purpose?

**Review by BlaineEXE — APPROVED** [review]
LGTM, just a question about spacing before merge.

File: .github/workflows/go-test.yaml (hunk 1 above)
Comment (subhamkrai) [reply .github/workflows/go-test.yaml:138]: no, space was missed added now

**Comment by subhamkrai** [comment]
> LGTM, just a question about spacing before merge.

yes, added the space

**Pull request merged**